invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/search/DeleteModal.js

```diff
@@ -0,0 +1,141 @@
+// This file is part of InvenioAdministration
+// Copyright (C) 2022 CERN.
+//
+// Invenio RDM Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Button, Icon, Message, Checkbox } from "semantic-ui-react";
+import { InvenioAdministrationActionsApi } from "@js/invenio_administration/api/actions";
+import { i18next } from "@translations/invenio_administration/i18next";
+import { Modal } from "semantic-ui-react";
+import { Trans } from "react-i18next";
+import isEmpty from "lodash/isEmpty";
+import { ErrorMessage } from "@js/invenio_administration/ui_messages/messages";
+import { NotificationContext } from "@js/invenio_administration/ui_messages/context";
+
+export class DeleteModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { loading: false, error: undefined, checked: false };
+  }
+
+  static contextType = NotificationContext;
+
+  cleanError = () => {
+    this.setState({ error: undefined });
+  };
+
+  handleOnButtonClick = async () => {
+    const { successCallback, resource, apiEndpoint, idKeyPath, toggleModal } =
+      this.props;
+    const { addNotification } = this.context;
+    try {
+      await InvenioAdministrationActionsApi.deleteResource(
+        resource,
+        apiEndpoint,
+        idKeyPath
+      );
+      toggleModal(false);
+      addNotification({
+        title: "Success",
+        content: "Resource was successfully deleted.",
+        type: "success",
+      });
+      successCallback();
+    } catch (e) {
+      this.setState({
+        error: { header: "Action error", content: e, id: e.code },
+      });
+    }
+  };
+
+  warningMessageCmp = () => {
+    const { title } = this.props;
+    const { checked } = this.state;
+    return (
+      <>
+        <Message
+          warning
+          icon="warning sign"
+          header={
+            <Trans
+              defaults="This will delete the set '{{title}}'."
+              values={{ title: title }}
+            />
+          }
+          content={i18next.t(
+            "Before deleting, make sure to alert all harvesters that this set will no longer be available."
+          )}
+        />
+        <Checkbox
+          className="ml-5"
+          value={true}
+          checked={checked}
+          onChange={(event, data) => {
+            this.setState({ checked: data.checked });
+          }}
+          label={i18next.t(
+            "I have alerted all harvesters that this set will no longer be available."
+          )}
```

Comment:
cna we place this JSX directly in the render method? what is the advantage of defining a separate method for the message?

Reply:
IMHO it's easier to read and follow the code, specially wen render methods tend to get quite big, but in this case we can keep it inside as it's relatively small, thanks!